### ISSUE TITLE
Use of "static" in callables is deprecated in PHP 8.2

### ISSUE THREAD

**Issue by jnoordsij** [FakerPHP/Faker#479]
### Summary

While running my PHPUnit test suite on the first PHP 8.2 alpha release, I noticed the following deprecation warning:  
`Use of "static" in callables is deprecated in /app/vendor/fakerphp/faker/src/Faker/Provider/Base.php on line 374`

### Versions

|                | Version |
|:---------------|:--------|
| PHP            | 8.2.0alpha1   |
| `fakerphp/faker`  | 1.19.0   |

**Comment by DvDty** [comment]
Yes, this usage falls under the deprecated callable patterns in version 8.2 More on that here, including two ways of fixing the issue: https://php.watch/versions/8.2/partially-supported-callable-deprecation#deprecated-patterns](https://php.watch/versions/8.2/partially-supported-callable-deprecation#deprecated-patterns).

I don't think there have been a discussion on whether or not `1.*` will support PHP 8.2, but since `2.x` is still [under discussion/development](https://github.com/FakerPHP/Faker/discussions/15), I would guess it will.

**Comment by pimjansen** [comment]
I think we will support it indeed however lets wait until the feature freeze which is at:

Jul 19 2022	Feature freeze

From there we can prep the needed steps

**Comment by chris-doehring** [comment]
Hello everyone,

I just created the following PR which ensures php 8.2 compatibility: https://github.com/FakerPHP/Faker/pull/528
As already mentioned in my PR, we have many of those deprecation errors with php 8.2 right now, but I think they probably cannot be fixed without causing a backward incompatible change. I think a better solution is to refactor those legacy Faker components into the new structure, so we will be compatible to php 9.0. It's not really necessary to fix those deprecation's right away, and I wouldn't consider them "bugs" either.
